docs/exit_validation.md

```diff
@@ -51,6 +50,8 @@ This is acceptable as long as the delay doesn't exceed `scheduled finalization t
 
 Conditions 3 and 4 can be represented jointly, by **continuous** (every child block) validation of the exits.
 If any exit is invalid and its finalization is near, then actions listed under 3 and 4 should take place.
+Let's call this joint representation of 3 and 4 `unchallenged_exit`.
```

Comment:
When the text says: `watcher must not allow spending and depositing` - is that prohibited on ChildChain as well? Or could a "hacker" exploit that assumption and allow spending/depositing from the watcher?

Reply:
When the text says: `(optional) If finalization of an invalid exit occurs, the watcher must prompt an exit without guarantees of recovery of funds` this basically means FIFO? Whoever is first to exit can get _money back_, if you're last, maybe there won't be anything left to exit with from the contract?

Reply:
This not allowing has nothing to do with the child chain. The Watcher protects its user, by not allowing certain actions under certain conditions. The child chain cannot possibly protect the user like that: it isn't trusted by the user.

Or, hmmm, maybe I don't get the question here... what would precisely such hacker do?

Reply:
I'm just exploring the possibility of a hacker cloning our repository, modifying the code by *allowing* spending/depositing in certain conditions and how that would affect the operators child chain...

Reply:
> When the text says: `(optional) If finalization of an invalid exit occurs, the watcher must prompt an exit without guarantees of recovery of funds` this basically means FIFO? 

FIFO is not 100% precise, as the Priority of exiting has richer logic. 

> Whoever is first to exit can get _money back_, if you're last, 

Again, not only "first to exit" but the exit priority counts here. It is impacted both by time exit started and age of utxo, see `tesuji_blockchain....` docs and others.

Having said that

> maybe there won't be anything left to exit with from the contract?

yep that can happen in this particular scenario (which we're avoiding by following protocol/Watching etc.)

Reply:
> I'm exploring the possibility of a hacker cloning our repository, modifying the code by _allowing_ spending/depositing in certain conditions and how that would affect the operators child chain...

If someone uses such modified code, anything can happen. In particular one can turn this particular function of the Watcher off and whoever uses that is exposed. Assuming the operator is using unmodified child chain code, it is never affected by such Watcher's operation.